src/lib/falcor/FalcorRouter.js

```diff
@@ -11,25 +28,169 @@ export default class FalcorRouter extends BaseRouter.createClass([
+            pathSet[2].forEach((field) => {
+              let ghostField = ghostify(field);
+              results.push({
+                path: ["articlesBySlug", article.slug, field],
+                value: article[ghostField]
+              });
+            });
+          });
+          resolve(results);
+        });
+      });
+    }
+  },
+  {
+    //Get custom article data from SQL database
+    route: "articlesBySlug[{keys:slugs}]['issue', 'category', 'description']",
+    get: (pathSet) => {
+      return new Promise((resolve, reject) => {
+        let query = {slug: {$in: pathSet.slugs}};
+        let projection = {_id: 0};
+        pathSet[2].forEach((field) => {
+          projection[field] = 1;
+        });
+        projection['slug'] = 1;
+        SQLArticleQuery(query, projection).then((data) => {
+          const results = [];
+          data.forEach((article) => {
+            pathSet[2].forEach((field) => {
+              results.push({
+                path: ["articlesBySlug", article.slug, field],
+                value: article[field]
+              });
+            });
+          });
+          resolve(results);
         });
+      });
+    }
+  },
+  {
+    // Get author information from article
+    route: "articlesBySlug[{keys:slugs}]['authors'][{integers:indices}]",
+    get: (pathSet) => {
+      console.log("here");
+      return new Promise((resolve, reject) => {
+        console.log("there");
+        let query = {slug: {$in: pathSet.slugs}};
+        let projection = {_id: 0, slug: 1, authors: 1};
+        SQLArticleQuery(query, projection).then((data) => {
+          console.log("made it and data coming here");
+          const results = [];
+          console.log(data);
+          data.forEach((article) => {
+            let authors = article.authors;
+            let indices = pathSet.indices;
+            for (let i = 0; i < indices.length; i++) {
+              let index = indices[i];
+              if (index >= authors.length) {
+                continue;
+              }
+              results.push({
+                path: ['articlesBySlug', article.slug, 'authors', index],
+                value: $ref(['authorsBySlug', authors[index]])
+              });
+            }
+          });
+          console.log("resolving");
+          console.log(results);
+          resolve(results);
+        });
+      });
+    }
+  },
+  {
+    // Get articles in chronological order by a consecutive range of indices where both min and max are multiples
+    // of the length of the range they span this is useful for when calling from index 0 to some element,
+    // or if calling pages one at a time and each page has the same number of elements. 
+    // Everything is always 0-based indexing
+    route: "articlesByTime['paginated'][{ranges:indices}]",
+    get: (pathSet) => {
+      if (!(pathSet.indices.length === 1 && (pathSet.indices[0].to+1)%(pathSet.indices[0].to-pathSet.indices[0].from+1) === 0)) {
```

Comment:
I think it would be better to have the developer just directly request a specific page #, and just return all posts in that page as an array, what do you think?